Content.Client/UserInterface/Controls/RadialMenu.cs

```diff
@@ -172,55 +180,66 @@ public void ReturnToPreviousLayer()
     }
 }
 
-[Virtual]
-public class RadialMenuButton : Button
+/// <summary>
+/// Special button for closing radial menu or going back between radial menu levels.
+/// Is looking like just <see cref="TextureButton "/> but considers whole space around
+/// itself (til radial menu buttons) as itself in case of clicking. Also considers all space
+/// outside of radial menu buttons as itself for clicking.
+/// </summary>
+public sealed class RadialMenuContextualCentralTextureButton : TextureButton
 {
-    /// <summary>
-    /// Upon clicking this button the radial menu will transition to the named layer
-    /// </summary>
-    public string? TargetLayer { get; set; }
 
-    /// <summary>
-    /// A simple button that can move the user to a different layer within a radial menu
-    /// </summary>
-    public RadialMenuButton()
+    /// <inheritdoc />
+    public RadialMenuContextualCentralTextureButton()
     {
-        OnButtonUp += OnClicked;
-    }
 
-    private void OnClicked(ButtonEventArgs args)
-    {
-        if (TargetLayer == null || TargetLayer == string.Empty)
-            return;
-
-        var parent = FindParentMultiLayerContainer(this);
-
-        if (parent == null)
-            return;
-
-        parent.TryToMoveToNewLayer(TargetLayer);
     }
 
-    private RadialMenu? FindParentMultiLayerContainer(Control control)
+    /// <summary>
+    /// Reference for container of radial menu.
+    /// Is required to properly consider radius of circles menu will draw.
+    /// </summary>
+    public RadialContainer? ActiveContainer { get; set; }
+
+    /// <inheritdoc />
+    protected override bool HasPoint(Vector2 point)
     {
-        foreach (var ancestor in control.GetSelfAndLogicalAncestors())
+        if (ActiveContainer == null)
         {
-            if (ancestor is RadialMenu)
-                return ancestor as RadialMenu;
+            return false;
         }
 
-        return null;
+        var cX = -Position.X + Parent!.Width / 2;
+        var cY = -Position.Y + Parent.Width / 2;
```

Comment:
This is almost certainly supposed to be Parent.Height?

Reply:
yeap, fixed that